src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -119,6 +119,11 @@ private void analyze(boolean includeRules) {
       LOG.warn("*                Use MSBuild 14 to get the best analysis results                 *");
       LOG.warn("* The use of MSBuild 12 or the sonar-runner to analyze C# projects is DEPRECATED *");
       LOG.warn("**********************************************************************************");
+
+      ImmutableMultimap<String, ActiveRule> activeRoslynRulesByPartialRepoKey = RoslynProfileExporter.activeRoslynRulesByPartialRepoKey(ruleProfile.getActiveRules());
+      if (activeRoslynRulesByPartialRepoKey.keySet().size() > 1) {
+        throw new IllegalArgumentException("Custom and 3rd party Roslyn analyzers are only by MSBuild 14. Either use MSBuild 14, or disable the custom/3rd party Roslyn analyzers in your quality profile.");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 205 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00103)